docs/concept-exercises.md

```diff
@@ -0,0 +1,74 @@
+
+For example, we might define a concept of "Classes" and provide a short introduction that explains what a class is, how it fits with objects, state, etc. We might include a link to a good article introducing OOP and classes. Individual tracks implementing an exercise on Classes can then include this introductory text, making any additions that explain their language-specific semantics and syntax.
+
+## Exercise Structure
+
+An exercise has the following files. In the browser they will show at the relevant times. When used via the CLI, the introduction and instructions will be concatenated along with the track's CLI instructions into a README.md, which will sit alongside a HINTS.md.
+
+### `.docs/introduction.md`
+
+This file provides an introduction to the concept. It should be explicit about what the exercise teaches and maybe provide a brief introduction to the concepts, but not give away so much that the user doesn't have to do any work to solve the exercise.
+
+See the C# floating-point-numbers exercise's [introduction.md file](./languages/csharp/concept-exercises/numbers-floating-point/.docs/introduction.md) for an example.
+
+### `.docs/instructions.md`
+
+This file provides instructions for the exercise. It should explicitly explain what the user needs to do (define a function with the signature `X.y(...)` that takes an A and returns a Z), and provide at least one example usage of that function. If there are multiple tasks within the exercise, it should provide an example of each.
+
+See the C# floating-point-numbers exercise's [instructions.md file](./languages/csharp/concept-exercises/numbers-floating-point/.docs/instructions.md) for an example.
+
+### `.docs/hints.md`
+
+If the user gets stuck, we will allow them to click a button requesting a hint, which shows this file. We will softly discourage them using it.
+
+The file should contain both general and task-specific "hints". These hints should be enough to unblock almost any user. They might link to the docs of the functions that need to be used.
+
+See the C# floating-point-numbers exercise's [hints.md file](./languages/csharp/concept-exercises/numbers-floating-point/.docs/hints.md) for an example.
+
+### `.docs/after.md`
+
+Once the user completes the exercise they will be shown this file, which gives them any bonus information or further reading about the concept taught.
+
+See the C# floating-point-numbers exercise's [after.md file](./languages/csharp/concept-exercises/numbers-floating-point/.docs/after.md) for an example.
+
+### `.docs/debug.md`
```

Comment:
I'll do a PR later to remove these here and document them somewhere else.

Reply:
Great. Thanks :)